examples/rbpf_plugin.rs

```diff
@@ -42,7 +42,7 @@ fn main() {
                     println!("JIT not supported on Windows");
                     return;
                 }
-                #[cfg(jit)] {
+                #[cfg(feature = "jit")] {
```

Comment:
Please squash these with the first commit.

Reply:
Ok that's done, would you say I should keep all changes that I add with my PR within a single commit so that it is easier to review? It is currently done like this

Reply:
No, please don't do a single commit for all the PR! Some of the steps are logically separated, like adding the `jit` feature flag and then handling the changes for no_std support.

What I would like is a set of logical, but clean commits. In a sense that each commit should contain a logical change, _and_ should be self-contained. There shouldn't be later commit with fixes for the first commits; instead, all “fixes” should be squashed into the commits they improve, to make sure each commit works on its own, and doesn't break the build.

Does it make sense?

Reply:
Yes that makes sense I will do my best to follow this practice. Thank you for the feedback, that's really helpful and I agree that it would make it much easier to review and then go through later on.

Reply:
You're welcome. Kernel docs give more guidance about this, if you wanted more:
- https://docs.kernel.org/process/submitting-patches.html#separate-your-changes
- https://docs.kernel.org/process/5.Posting.html#patch-preparation

The objective is to be able to review logical changes independently, and efficiently. One motivation to preserve the builds through each commit is to be able to `git bisect` efficiently, if needed.

I'll review the rest of your changes soon. Thanks a lot!

Reply:
Ok thank you for the additional pointers, I've read through those, that's really useful.
Now that I think about this, the jit flag complicates things quite a bit and I think I can get around that using combinations of cfg not(windows) and feature = "std" (see replies above). In this case I think I will open up a new PR and apply the current changes there in a clean way in a single commit. The thing with adding no_std is that I can't really do it half way through so (it needs to be 'atomic') otherwise it won't compile without std. In which case I think it is a single logical change and could be put into a single commit. 

Thank you for all of your comments and suggestions in this thread. Is it ok if I open the new clean PR and have the final version of changes there without all of this intermediate changes that happened here? Or should I continue here so that our conversation gets preserved?

Reply:
You're free to open a new PR, but I don't really see the point to be honest, it's perfectly fine to update and force-push here.

I'd still prefer having several, smaller commits. For example, adding the `std` feature to the Cargo.toml and using it to gate the related code in addition to `windows` is a logical change and doesn't break the build, even if the `std` feature has not been implemented yet. I'm pretty sure some other parts can be their own commits, too. Don't lose too much time on that, but generally speaking, just keep in mind that smaller commits are easier to review and understand in the future.

Reply:
Ok sounds good, thank you for your answer. I do agree that opening up a new RP could potentially create additional mess. I will force-push the clean changes to this one then and try to break them up into separate logical commits